### PR TITLE
DEVPROD-5498 Do not consider elapased communication time during group teardown

### PR DESCRIPTION
**Pull request by hadjri**
DEVPROD-5498

### Description
Currently, hosts can get hit with an idle timeout due to lack of communication with the Evergreen app server if their teardown group is sufficiently long, because the teardown group happens after task completion, so no heartbeat signals are sent during that period. 

Since the change introduced in #7635 adds protections against long-running teardown groups, it should be sufficient to skip the last communicated time in the idle host check _if_ a host is actively tearing down a task group.
### Testing
[Tested in staging](https://spruce-staging.corp.mongodb.com/version/6740dfc0f354c60008e9cf6a/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) (executions 3 vs 4 for reference) and confirmed that without the change, a host that just ran a task group with a long teardown group is unable to pick up more tasks afterwards because it immediately gets marked by the idle termination job, whereas after the change, said host is able to continue picking up tasks.

### CODE REVIEW

File: model/host/host.go [hunk 1]
```diff
@@ -2169,6 +2169,12 @@ func (h *Host) Replace(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (hadjri) [reply model/host/host.go:2169]: Sure. For reference task1+task2 are part of a group, and task3 is a standalone task.

Execution 3 (with change):

task1+task2 complete on host [i-0ba489af415acc35e](https://spruce-staging.corp.mongodb.com/host/i-0ba489af415acc35e), and then that same host is able to immediately pick up task 3 afterwards.

Execution 4 (without change):
task1+task2 complete on host [i-0e34ab569dac24127](https://spruce-staging.corp.mongodb.com/host/i-0e34ab569dac24127), which gets hit with an idle timeout due to the task group's teardown group taking too long, and it decommissions. task3 then comes up on a new host.

[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/host/host.go:2169]: I only see task three running on [i-0ba489af415acc35e](https://spruce-staging.corp.mongodb.com/host/i-0ba489af415acc35e), I don't see it running task 1, 2 and 3.

[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/host/host.go:2169]: (I'm guessing you meant [i-059849d0ff65edfc0]( https://spruce-staging.corp.mongodb.com/host/i-059849d0ff65edfc0))

**Review by malikchaya2 — APPROVED** [review]

File: model/host/host.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hadjri) [reply model/host/host.go:2169]: Yes, I meant [i-059849d0ff65edfc0](https://spruce-staging.corp.mongodb.com/host/i-059849d0ff65edfc0), my bad

**Pull request merged**